pennylane/ops/qubit/hamiltonian.py

```diff
@@ -552,14 +558,19 @@ def __add__(self, H):
         if isinstance(H, Hamiltonian):
             coeffs = qml.math.concatenate([self_coeffs, copy(H.coeffs)], axis=0)
             ops.extend(H.ops.copy())
-            return qml.Hamiltonian(coeffs, ops, simplify=True)
+            new_hamiltonian = qml.Hamiltonian(coeffs, ops, simplify=True)
+            qml.QueuingContext.safe_update_info(self, owner=new_hamiltonian)
+            qml.QueuingContext.safe_update_info(H, owner=new_hamiltonian)
+            return new_hamiltonian
```

Comment:
Can you hint at what is happening here? I see that you manipulate ownership...

Reply:
In order for the two initial hamiltonians to avoid getting processed in the tape, we need to give them owners.